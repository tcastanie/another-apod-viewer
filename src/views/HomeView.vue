<template>
  <ApodViewer />
</template>

<script>
import ApodViewer from "../components/ApodViewer.vue";
export default {
  components: { ApodViewer },
};
</script>
