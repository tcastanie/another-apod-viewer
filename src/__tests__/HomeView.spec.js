import { describe, it, expect, beforeEach } from "vitest";
import { mount } from "@vue/test-utils";
import { createRouter, createWebHistory } from "vue-router";
import { routes } from "@/router/index.js";
import HomeView from "@/views/HomeView.vue";

let router;
beforeEach(async () => {
  router = createRouter({
    history: createWebHistory(),
    routes,
  });
});

describe("HomeView", async () => {
  it("renders properly", async () => {
    router.push("/");
    await router.isReady();
    const wrapper = mount(HomeView, {
      global: {
        plugins: [router],
        stubs: {
          "apod-viewer": true,
        },
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("shows correct link", async () => {
    router.push("/");
    await router.isReady();
    const wrapper = mount(HomeView, {
      global: {
        plugins: [router],
        stubs: {
          "apod-viewer": true,
        },
      },
    });
    const yesterday = new Date();
    yesterday.setDate(yesterday.getDate() - 1);
    const yesterdayDate = yesterday.toISOString().split("T")[0];
    expect(wrapper.find("a").attributes("href")).toContain(yesterdayDate);
  });
});
