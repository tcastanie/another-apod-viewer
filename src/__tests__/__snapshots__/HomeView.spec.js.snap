// Vitest Snapshot v1

exports[`HomeView > renders properly 1`] = `
"<apod-viewer-stub></apod-viewer-stub>
<a href=\\"/2022-08-20\\" class=\\"\\">Previous</a>"
`;
